contracts/RibbonVaultAssetOracle.sol

```diff
@@ -0,0 +1,239 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/utils/math/SafeMathUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+
+import "./interfaces/IPriceOracle.sol";
+import "./interfaces/IBasePriceOracle.sol";
+import "./interfaces/ICToken.sol";
+import "./interfaces/ICErc20.sol";
+import "./interfaces/IAggregatorV3Interface.sol";
+import "./interfaces/ILiquidityGauge.sol";
+import "./interfaces/IRibbonVault.sol";
+import {DSMath} from "./libraries/DSMath.sol";
+
```

Comment:
[tests](https://github.com/ribbon-finance/governance/blob/main/test/borrow-oracle/FuseOracle.ts) for this